core/controllers/acl_decorators.py

```diff
@@ -2919,9 +2919,31 @@ def test_can_edit(
             raise self.NotFoundException
         if role_services.ACTION_EDIT_ANY_QUESTION in self.user.actions:
             return handler(self, question_id, **kwargs)
-        else:
+        if (role_services.ACTION_EDIT_QUESTION_IN_MANAGED_TOPIC
+        in self.user.actions):
+            skills = question_services.get_skills_linked_to_question(
+                question_id)
+            if not skills:
+                raise self.UnauthorizedUserException(
+                    'You do not have credentials to edit this question.')
+            skill_ids = set(skill.id for skill in skills)
+            managed_topic_ids = [
+                    rights.id for rights in
+                    topic_fetchers.get_topic_rights_with_user(self.user_id)]
+            if not managed_topic_ids:
+                raise self.UnauthorizedUserException(
+                    'You do not have credentials to edit this question.')
+            for topic_id in managed_topic_ids:
+                topic = topic_fetchers.get_topic_by_id(topic_id, strict=False)
```

Comment:
This needs to be done in a single get_multi rather than N individual gets, which takes too long.

Reply:
Done!